staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +61,167 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 			})
 		}
 
-		pod.Spec.HostIPC = true
 		pod.Spec.HostNetwork = true
 		pod.Spec.HostPID = true
+		pod.Spec.HostIPC = true
 		return nil
 	default:
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func (p *generalProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	switch t := target.(type) {
+	case *corev1.Pod:
+		if t == pod {
+			// For ephemeral container: sets SYS_PTRACE in ephemeral container
+			for i, c := range pod.Spec.EphemeralContainers {
```

Comment:
That would be more simplified. Don't we manage debug container whose name is `containerName` which we are looking for and we know that this debug container does not contain any SecurityContext?

```go
			if c.Name != containerName {
				continue
			}
			// For ephemeral container: sets SYS_PTRACE in ephemeral container
			pod.Spec.EphemeralContainers[i].SecurityContext = &corev1.SecurityContext{
				Capabilities: &corev1.Capabilities{
					Add: []corev1.Capability{"SYS_PTRACE"},
				},
			}
```

Reply:
and we can do the same all the other profiles.

Reply:
Yeah, you're right. I have pushed the code update in ad29d22cb926929d8bb51f3b12cabdd5694711ea.

However, the reason why I did this as well as explicitly set pod's HostNework, HostPID and HostIPC fields to false in the case of node debugging is that I wanted to avoid a coupling between:

1. the way that the debug container or debug pod (in the case of node debugging) is constructed by the ProfilesApplier's callers, and
2. the ProfileApplier implementations themselves